### PR TITLE
Add sample tests for MockTrigger

### PR DESCRIPTION
**Pull request by mds1**
MockTrigger is now slightly more complex to accompany a sample set of tests that provide a good starting point for most trigger development and testing

### CODE REVIEW

File: contracts/MockTrigger.sol [hunk 1]
```diff
@@ -1,26 +1,55 @@
...
+
+    // Verify market is not already triggered.
+    require(!isMarketTriggered(), "Already triggered");
+  }
+
+  /**
+   * @notice Special function for this mock trigger to set whether or not the trigger should toggle
+   */
+  function setShouldToggle(bool _shouldToggle) external {
+    shouldToggle = _shouldToggle;
```
Comment (apbendi) [inline contracts/MockTrigger.sol:29]: Consider adding a `require(!isTriggered)` with a useful error message as a convenience to the developer. If this method is ever called after `isTriggered`, the dev probably messed something up and it's better to let them know.

File: contracts/MockTrigger.sol [hunk 2]
```diff
@@ -1,26 +1,55 @@
...
+    return shouldToggle;
+  }
 
   /**
    * @notice Checks trigger condition, sets isTriggered flag to true if condition is met, and
    * returns the trigger status
    * @dev For this mock trigger, there is no condition to check, so it just returns isTriggered
    */
   function checkAndToggleTrigger() external override returns (bool) {
+    // Short circuit if trigger already toggled
```
Comment (apbendi) [inline contracts/MockTrigger.sol:45]: No action required here, just a stylistic comment. For a case like this, where there are multiple paths but none of them are clear "happy" path (i.e. you're not checking error or validation conditions), I'd probably use a conventional if-if else-else statement to signal as much to other readers of the code.

**Review by apbendi — APPROVED** [review]
Looks good overall! One suggested change for dev convenience only. Otherwise to merge 👍

File: contracts/MockTrigger.sol (hunk 1 above)
Comment (mds1) [reply contracts/MockTrigger.sol:29]: Done in 004ecce97745bd204e600454e08f32c7351527d0

File: contracts/MockTrigger.sol (hunk 2 above)
Comment (mds1) [reply contracts/MockTrigger.sol:45]: Noted! 👌

**Pull request merged**